pkg/services/queryhistory/query_history_service.go

```diff
@@ -0,0 +1,62 @@
+package queryhistory
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+func ProvideService(cfg *setting.Cfg, sqlStore *sqlstore.SQLStore) *QueryHistoryService {
```

Comment:
You could also pass `routing.RouteRegister` so that you could register the endpoints here as it happens with the [libraryelements](https://github.com/grafana/grafana/blob/50fabe8a87028646b0a39a18ae45566388b3a737/pkg/services/libraryelements/libraryelements.go#L20)
In that way, in combination with the following comment: the endpoints will be available only if the feature is enabled (and eventually you will no more need this check:
https://github.com/grafana/grafana/blob/958f15056d630a22542bb3f83c69df3f873ace4f/pkg/services/queryhistory/query_history_service.go#L32-L34)

Reply:
Great feedback, I have updated this so the routes only register when the feature is enabled.